src/Microsoft.AspNetCore.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -16,16 +16,22 @@ public class StreamSocketOutput : ISocketOutput
         private static readonly byte[] _endChunkBytes = Encoding.ASCII.GetBytes("\r\n");
         private static readonly byte[] _nullBuffer = new byte[0];
 
+        private readonly string _connectionId;
         private readonly Stream _outputStream;
         private readonly MemoryPool _memory;
+        private readonly IKestrelTrace _logger;
         private MemoryPoolBlock _producingBlock;
 
+        private bool _canWrite = true;
+
         private object _writeLock = new object();
 
-        public StreamSocketOutput(Stream outputStream, MemoryPool memory)
+        public StreamSocketOutput(string connectionId, Stream outputStream, MemoryPool memory, IKestrelTrace logger)
         {
+            _connectionId = connectionId;
             _outputStream = outputStream;
             _memory = memory;
+            _logger = logger;
         }
 
         public void Write(ArraySegment<byte> buffer, bool chunk)
```

Comment:
I think this method needs to be changed too. We should log and swallow exceptions from _outputStream.Write and we should ignore zero length writes.

While we're at it ChunkedResponseTests should be run with the NoOpConnectionFilter like the EngineTests are.